util/src/main/java/io/camunda/zeebe/util/buffer/BufferUtil.java

```diff
@@ -114,6 +114,14 @@ public static DirectBuffer cloneBuffer(
     }
   }
 
+  public static DirectBuffer trimBufferIfExceeds(final DirectBuffer src, final int maxLength) {
```

Comment:
I am not sure if this method name exactly explains the logic since we trim but also do a clone inside. But I couldn't come up with a better name. Please feel free to offer if you can think of a better one.

Reply:
How about `cap` or something similar to `subString`

Reply:
Voting for `truncate`.
 `cap` is confusing IMO because buffers already have a `cap` or `capacity` which is a different concept.

Reply:
To me, `truncate` sounds similar to `trim` which doesn't indicates the cloning part. `subString` reminds of method without side effect but not sure if we should say string because the util class doesn't seem to be dedicated to string operations only. Maybe `subIfExceeds`? :thinking:

Reply:
Do you need the copy? I think 
```java
  public static DirectBuffer trimBufferIfExceeds(final DirectBuffer src, final int maxLength) {
    return new UnsafeBuffer(src, 0, maxLength);
  }
```
would work as well. It's returning a "new" buffer that simply hides the excess data and doesn't create a copy.

Reply:
@oleschoenburg I checked it and this is definitely a better one. I will go with that option. Thanks!